subprojects/soak/src/integTest/groovy/org/gradle/file/archive/TarFileTreeSoakTest.groovy

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.file.archive
+
+import org.gradle.api.file.EmptyFileVisitor
+import org.gradle.api.file.FileVisitDetails
+import org.gradle.api.internal.file.MaybeCompressedFileResource
+import org.gradle.api.internal.file.TestFiles
+import org.gradle.api.internal.file.archive.TarFileTree
+import org.gradle.api.provider.Provider
+import org.gradle.api.resources.internal.LocalResourceAdapter
+import org.gradle.cache.internal.TestCaches
+import org.gradle.test.fixtures.file.TestFile
+import org.gradle.test.fixtures.file.TestNameTestDirectoryProvider
+import org.gradle.util.TestUtil
+import org.junit.Rule
+import spock.lang.Issue
+import spock.lang.Specification
+
+import java.nio.file.Files
+import java.nio.file.StandardCopyOption
+
+import static org.gradle.api.internal.file.TestFiles.directoryFileTreeFactory
+import static org.gradle.api.internal.file.TestFiles.fileHasher
+import static org.gradle.api.internal.file.TestFiles.fileSystem
+import static org.gradle.util.internal.WrapUtil.toList
+
+class TarFileTreeSoakTest extends Specification {
```

Comment:
❓ Why not reuse `AbstractIntegrationSpec` here? That would allow to get rid of the custom `temporaryFolder` and have the test behave the same as the other ones wrt. temporary files.

Reply:
My thinking was to not bring in a bunch of unused functionality until not really needing it, but I don't feel strongly about it. Will change it.